workflow/controller/operator.go

```diff
@@ -1128,10 +1128,19 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 	}
 
 	newTmplCtx, basedTmpl, err := woc.getResolvedTemplate(node, orgTmpl, tmplCtx, args)
+
 	if err != nil {
 		return woc.initializeNodeOrMarkError(node, nodeName, wfv1.NodeTypeSkipped, orgTmpl, boundaryID, err), err
 	}
 
+	if woc.wf.Spec.HasPodSpecPatch() {
+		err = util.PodSpecPatchMerge(woc.wf, basedTmpl)
```

Comment:
Can we move this merge into createWorkflowPod() instead of here. I don't think manipulating the template at this level is appropriate.